runner/run.go

```diff
@@ -259,18 +259,18 @@ func Run(ctx context.Context, wg *sync.WaitGroup, globalCollectionOpts state.Col
 	}
 
 	if hasAnyActivityEnabled {
-		schedulerGroups["activity"].Schedule(ctx, func(ctx context.Context) {
+		scheduler.TenSecond.Schedule(ctx, func(ctx context.Context) {
 			wg.Add(1)
 			CollectActivityFromAllServers(ctx, servers, globalCollectionOpts, logger)
 			wg.Done()
 		}, logger, "activity snapshot of all servers")
 	}
 
-	schedulerGroups["query_stats"].ScheduleSecondary(ctx, func(ctx context.Context) {
+	scheduler.OneMinute.ScheduleSecondary(ctx, func(ctx context.Context) {
 		wg.Add(1)
 		GatherQueryStatsFromAllServers(ctx, servers, globalCollectionOpts, logger)
 		wg.Done()
-	}, logger, "high frequency query statistics of all servers", schedulerGroups["stats"])
+	}, logger, "high frequency query statistics of all servers", scheduler.TenMinute)
```

Comment:
I wonder if this could benefit from a comment, since we're not losing the context that this the primary group this is scheduled in relation to, is the `CollectAllServers` stats collection. Maybe:

```suggestion
	}, logger, "high frequency query statistics of all servers",
	scheduler.TenMinute // Skip at every 10 minute mark, when we collect full snapshots with query text
      )
```

Reply:
I wonder if the the schedule should be moved earlier in the argument list since it's fairly important. Then the comment can be above the function call:
```go
// Query stats are captured every minute, except for minute 10 when full snapshot collection takes over
scheduler.OneMinute.ScheduleSecondary(ctx, scheduler.TenMinute, func(ctx context.Context) {
	wg.Add(1)
	GatherQueryStatsFromAllServers(ctx, servers, globalCollectionOpts, logger)
	wg.Done()
}, logger, "high frequency query statistics of all servers")

Reply:
Yeah, good idea, I think moving it earlier works.